bindings/cs/MongoDB.Crypt.Test/BasicTests.cs

```diff
@@ -212,6 +218,36 @@ public void EncryptExplicit()
             }
         }
 
+        [Fact]
+        public void EncryptExplicitWithLocalAndAwsCredentials()
+        {
+            var keyDoc = ReadJsonTestFile("key-document.json");
+            Guid key = keyDoc["_id"].AsGuid;
+
+            BsonDocument doc = new BsonDocument()
+            {
+                {  "v" , "hello" },
+            };
+
+            var testData = BsonUtil.ToBytes(doc);
+
+            byte[] encryptedBytes;
+            using (var cryptClient = CryptClientFactory.Create(CreateOptions(true)))
+            using (var context = cryptClient.StartExplicitEncryptionContext(key, EncryptionAlgorithm.AEAD_AES_256_CBC_HMAC_SHA_512_Random, testData))
+            {
+                var (encryptedBinary, encryptedDocument) = ProcessContextToCompletion(context);
+                encryptedBytes = encryptedBinary.ToArray(); // need to copy bytes out before the context gets destroyed
+            }
+
+            using (var cryptClient = CryptClientFactory.Create(CreateOptions(true)))
+            using (var context = cryptClient.StartExplicitDecryptionContext(encryptedBytes))
+            {
+                var (decryptedResult, _) = ProcessContextToCompletion(context);
+
+                decryptedResult.ToArray().Should().Equal(testData);
+            }
```

Comment:
This looks good to me: we're not encrypting/decrypting with the local key but I don't think there's an easy way to create that test since we'd need the mock replies, but the driver side tests should cover this case